streaming-binlog/src/main/java/org/openmrs/analytics/GetUuidUtil.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+
+import java.beans.PropertyVetoException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Map;
+
+public class GetUuidUtil {
+	
+	private JdbcConnectionUtil jdbcConnectionUtil;
+	
+	public GetUuidUtil(JdbcConnectionUtil jdbcConnectionUtil) {
+		this.jdbcConnectionUtil = jdbcConnectionUtil;
+	}
+	
+	public String getUuid(String parentTable, String parentForeignKey, String childPrimaryKey, Map payload)
```

Comment:
am using the payload we need to make sure that the value is exactly the child primarykey from the payload generated  otherwise with out that the sql statement returns this `java.sql.SQLSyntaxErrorException: Unknown column 'patient_id' in 'where clause'` thats why we fetch the child primary key exactly from the payload of the child table

Reply:
Sorry let me clarify my previous comment as it was not accurate: From `childPrimaryKey` and `payload` all you need is `payload.get(childPrimaryKey)`, correct? So what I am suggesting is instead of passing two args just pass one String; especially the `Map` arg is problematic here and makes the API for this method unnecessarily complex. I would even go one step further and suggest to completely drop the notion of parent/child in this API. So the function signature would be:
```
public String getUuid(String table, String keyColumn, String keyValue)
```
All this function does is to retrieve a row from a given table using a key column, right?

When you call this function in `FhirConverter` it would be:
```
uuid = getUuidUtil.getUuid(config.getParentTable(), config.getParentForeignKey(), payload.get(config.getChildPrimaryKey());
```

Reply:
sure, fixed it